apps/meteor/server/models/raw/LivechatDepartmentAgents.ts

```diff
@@ -395,6 +396,39 @@ export class LivechatDepartmentAgentsRaw extends BaseRaw<ILivechatDepartmentAgen
 	): FindCursor<ILivechatDepartmentAgents> {
 		return this.find({ agentId: { $in: agentsIds }, departmentId }, options);
 	}
+
+	findDepartmentsOfAgent(agentId: string, enabled = false): AggregationCursor<ILivechatDepartmentAgents & { departmentName: string }> {
+		return this.col.aggregate<ILivechatDepartmentAgents & { departmentName: string }>([
+			{
+				$match: {
+					agentId,
+					...(enabled && { departmentEnabled: true }),
```

Comment:
Can we have a compound index to cover this new query? I believe it's worth it since the ones we currently have only cover each restriction individually but not them together